### PR TITLE
e2e: successful test where the channel is upgraded and fee middleware is enabled

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

~~Work in progress: I need a new docker image from the relayer that Luca is building as we speak.~~

Test passing [here](https://github.com/cosmos/ibc-go/actions/runs/7186712486/job/19572800383?pr=5358). 🥳 

closes: #5357

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/go.mod [hunk 1]
```diff
@@ -12,20 +12,20 @@ require (
 	github.com/cosmos/ibc-go/modules/light-clients/08-wasm v0.0.0-00010101000000-000000000000
 	github.com/cosmos/ibc-go/v8 v8.0.0
 	github.com/docker/docker v24.0.7+incompatible
-	github.com/strangelove-ventures/interchaintest/v8 v8.0.0
+	github.com/strangelove-ventures/interchaintest/v8 v8.0.1-0.20231206154526-2f014d308bea
```
Comment (crodriguezvega) [inline e2e/go.mod:15]: I needed to bump interchaintest to get this [PR](https://github.com/strangelove-ventures/interchaintest/pull/883) that added support for a new field (chain `type`) that has been added to hermes config.toml.

File: e2e/tests/core/04-channel/upgrades_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		version, err := feetypes.MetadataFromVersion(channel.Version)
+		s.Require().NoError(err)
+		s.Require().Equal(feetypes.Version, version.FeeVersion, "the channel version did not include ics29")
+
+		// extra check
+		feeEnabled, err := s.QueryFeeEnabledChannel(ctx, chainB, channelB.PortID, channelB.ChannelID)
+		s.Require().NoError(err)
+		s.Require().Equal(true, feeEnabled)
+	})
+}
```
Comment (crodriguezvega) [inline e2e/tests/core/04-channel/upgrades_test.go:117]: I could try to send some incentivised packets, but I thought that the checks above were already enough.

File: e2e/testsuite/testsuite.go [hunk 3]
```diff
@@ -15,6 +15,8 @@ import (
 	testifysuite "github.com/stretchr/testify/suite"
 	"go.uber.org/zap"
 
+	sdkmath "cosmossdk.io/math"
```
Comment (crodriguezvega) [inline e2e/testsuite/testsuite.go:18]: The changes in this filed are required because of the bump of interchaintest.

Comment (damiannolan) [reply e2e/testsuite/testsuite.go:18]: looks like there's a compiler error in "build e2e" workflow from this atm 👀

Comment (crodriguezvega) [reply e2e/testsuite/testsuite.go:18]: Ay, I will check!

**Comment by crodriguezvega** [comment]
Grandpa tests are failing, need to check what's going on there now.

File: e2e/sample.config.yaml [hunk 4]
```diff
@@ -30,7 +30,7 @@ relayers:
     tag: "latest"
   - id: hyperspace
     image: ghcr.io/misko9/hyperspace
-    tag: "timeout"
+    tag: "20231122v39"
```
Comment (damiannolan) [inline e2e/sample.config.yaml:33]: :D

**Review by damiannolan — APPROVED** [review]
Awesome work @crodriguezvega! Thanks

**Pull request merged**